File: fstp-core-bo/resource/com/purefun/fstp/core/bo/otw/MarketBarBO_OTW.java
package com.purefun.fstp.core.bo.otw;

import com.purefun.fstp.core.bo.MarketBarBO;
import com.purefun.fstp.core.bo.pro.MarketBarBO_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fstp.core.bo.commom.ICommom_OTW;
import com.google.protobuf.Any;

public class MarketBarBO_OTW implements ICommom_OTW {
    MarketBarBO_PRO.MarketBarBO.Builder builder = null;
    MarketBarBO bo = null;

    public MarketBarBO_OTW() {
        builder = MarketBarBO_PRO.MarketBarBO.newBuilder();
        bo= new MarketBarBO();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public MarketBarBO_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = MarketBarBO_PRO.MarketBarBO.newBuilder();
        bo= new MarketBarBO();
        MarketBarBO_PRO.MarketBarBO receive = MarketBarBO_PRO.MarketBarBO.parseFrom(message);
        setProduct_id(receive.getProductId());
        setExch(receive.getExch());
        setDate(receive.getDate());
        setHigh(receive.getHigh());
        setLow(receive.getLow());
        setOpen(receive.getOpen());
        setClose(receive.getClose());
        setChange(receive.getChange());
        setVolume(receive.getVolume());
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
    }

    public MarketBarBO_OTW(MarketBarBO bofrom){
        builder = MarketBarBO_PRO.MarketBarBO.newBuilder();
        bo= new MarketBarBO();
        setProduct_id(bofrom.product_id);
        setExch(bofrom.exch);
        setDate(bofrom.date);
        setHigh(bofrom.high);
        setLow(bofrom.low);
        setOpen(bofrom.open);
        setClose(bofrom.close);
        setChange(bofrom.change);
        setVolume(bofrom.volume);
        setUuid(bofrom.uuid);
        setBoid(bofrom.boid);
        setDestination(bofrom.destination);
    }

    public byte[] serial() {
        return builder.build().toByteArray();
    }

    @Override
    public com.google.protobuf.GeneratedMessageV3.Builder getBuilder() { 
        return builder;
    }

    @Override
    public MarketBarBO getBo() { 
        return bo;
    }

    public java.lang.String getProduct_id() {
        return builder.getProductId();
    }

    public void setProduct_id(java.lang.String product_id) {
        bo.product_id = product_id;
        builder.setProductId(product_id);
    }

    public java.lang.String getExch() {
        return builder.getExch();
    }

    public void setExch(java.lang.String exch) {
        bo.exch = exch;
        builder.setExch(exch);
    }

    public java.lang.String getDate() {
        return builder.getDate();
    }

    public void setDate(java.lang.String date) {
        bo.date = date;
        builder.setDate(date);
    }

    public double getHigh() {
        return builder.getHigh();
    }

    public void setHigh(double high) {
        bo.high = high;
        builder.setHigh(high);
    }

    public double getLow() {
        return builder.getLow();
    }

    public void setLow(double low) {
        bo.low = low;
        builder.setLow(low);
    }

    public double getOpen() {
        return builder.getOpen();
    }

    public void setOpen(double open) {
        bo.open = open;
        builder.setOpen(open);
    }

    public double getClose() {
        return builder.getClose();
    }

    public void setClose(double close) {
        bo.close = close;
        builder.setClose(close);
    }

    public double getChange() {
        return builder.getChange();
    }

    public void setChange(double change) {
        bo.change = change;
        builder.setChange(change);
    }

    public double getVolume() {
        return builder.getVolume();
    }

    public void setVolume(double volume) {
        bo.volume = volume;
        builder.setVolume(volume);
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public String toString() {
        return "MarketBarBO_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "product_id = " + getProduct_id() +"," +
            "exch = " + getExch() +"," +
            "date = " + getDate() +"," +
            "high = " + getHigh() +"," +
            "low = " + getLow() +"," +
            "open = " + getOpen() +"," +
            "close = " + getClose() +"," +
            "change = " + getChange() +"," +
            "volume = " + getVolume() +"," +
         "]";
    }
}

File: fstp-python-2.7/src/com/purefun/fstp/core/bo/otw/MarketBarBO_OTW.py
from src.core.common.ICommon_OTW import ICommon_OTW
from src.com.purefun.fstp.core.bo.pro import MarketBarBO_pb2
from src.com.purefun.fstp.core.bo.model.MarketBarBO import MarketBarBO

class MarketBarBO_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = MarketBarBO_pb2.MarketBarBO()
        self._bo = MarketBarBO()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.product_id = self._bo.product_id
        self._bo_pro.exch = self._bo.exch
        self._bo_pro.date = self._bo.date
        self._bo_pro.high = self._bo.high
        self._bo_pro.low = self._bo.low
        self._bo_pro.open = self._bo.open
        self._bo_pro.close = self._bo.close
        self._bo_pro.change = self._bo.change
        self._bo_pro.volume = self._bo.volume
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination

    def __setDataFromPB(self):
        self._bo.product_id = self._bo_pro.product_id
        self._bo.exch = self._bo_pro.exch
        self._bo.date = self._bo_pro.date
        self._bo.high = self._bo_pro.high
        self._bo.low = self._bo_pro.low
        self._bo.open = self._bo_pro.open
        self._bo.close = self._bo_pro.close
        self._bo.change = self._bo_pro.change
        self._bo.volume = self._bo_pro.volume
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getProduct_id(self):
        return self._bo.product_id

    def setProduct_id(self, product_id):
        self._bo.product_id = product_id
        self._bo_pro.product_id = product_id

    def getExch(self):
        return self._bo.exch

    def setExch(self, exch):
        self._bo.exch = exch
        self._bo_pro.exch = exch

    def getDate(self):
        return self._bo.date

    def setDate(self, date):
        self._bo.date = date
        self._bo_pro.date = date

    def getHigh(self):
        return self._bo.high

    def setHigh(self, high):
        self._bo.high = high
        self._bo_pro.high = high

    def getLow(self):
        return self._bo.low

    def setLow(self, low):
        self._bo.low = low
        self._bo_pro.low = low

    def getOpen(self):
        return self._bo.open

    def setOpen(self, open):
        self._bo.open = open
        self._bo_pro.open = open

    def getClose(self):
        return self._bo.close

    def setClose(self, close):
        self._bo.close = close
        self._bo_pro.close = close

    def getChange(self):
        return self._bo.change

    def setChange(self, change):
        self._bo.change = change
        self._bo_pro.change = change

    def getVolume(self):
        return self._bo.volume

    def setVolume(self, volume):
        self._bo.volume = volume
        self._bo_pro.volume = volume

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def toString(self):
        return "MarketBarBO_OTW ["+"product_id = " + self.getProduct_id() +"," +"exch = " + self.getExch() +"," +"date = " + self.getDate() +"," +"high = " + str(self.getHigh()) +"," +"low = " + str(self.getLow()) +"," +"open = " + str(self.getOpen()) +"," +"close = " + str(self.getClose()) +"," +"change = " + str(self.getChange()) +"," +"volume = " + str(self.getVolume()) +"," +"uuid = " + self.getUuid() +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + self.getDestination() +"," +"]"
